code/game/area/areas.dm

```diff
@@ -493,8 +493,8 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 
 /area/Entered(atom/movable/M)
 	set waitfor = FALSE
-	SEND_SIGNAL(src, COMSIG_AREA_ENTERED, M)
-	SEND_SIGNAL(M, COMSIG_ENTER_AREA, src) //The atom that enters the area
+	SEND_SIGNAL(src, COMSIG_AREA_ENTERED, M, src)
+	SEND_SIGNAL(M, COMSIG_ENTER_AREA, src, M) //The atom that enters the area
```

Comment:
Dude, the first argument of the macro is also the first in the signal procs args, this is redundant.